code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -54,3 +54,108 @@
 	max_charges = 10
 	recharge_rate = 2
 	no_den_usage = 1
+
+
+// Lord Staves
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant
+	name = "staff of revenant"
+	desc = "A cursed artifact that starts off weak, but you can drain the souls of dead bodies in order to make it more powerful! Activate the staff in hand to see how many souls you have and, if you have enough, make your staff stronger."
+	fire_sound = "sound/magic/WandODeath.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/staffofrevenant
+	icon_state = "staffofrevenant"
+	item_state = "staffofrevenant"
+	w_class = 4
+	max_charges = 1
+	recharge_rate = 10
+	no_den_usage = 1
+	var/revenant_level = 0
+	var/revenant_damage = 20 // doesn't actually change the damage, only used for the inhand activation
+	var/revenant_souls = 0
+	var/list/drained_mobs = list() //Cannot harvest the same mob twice
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant/attack(mob/living/carbon/human/target, mob/living/user)
+	if(target.stat & DEAD)
+		if(!(target in drained_mobs))
+			playsound(src,'sound/magic/Staff_Chaos.ogg',40,1)
+			user.visible_message("<font color=purple>[user] drains [target] their soul with [src]!</font>", "<span class='notice'>You use [src] to drain [target]'s soul, empowering your weapon!</span>")
+			revenant_souls++
+			drained_mobs.Add(target)
+		else
+			user << "<span class='warning'>[target]'s soul is dead and empty.</span>"
+			return
+	..()
+
+obj/item/weapon/gun/magic/staff/staffofrevenant/attack_self(mob/living/user)
+	if(revenant_level == 0)
+		if(revenant_souls >= 5)
+			user << "<font color=purple>As you focus on the staff, you witness the crystal eminating a bright shine, before receeding again. The staff hums at an eerie tone, and has managed to become much stronger...</font>"
+			max_charges = 2
+			charges = 2
+			recharge_rate = 9
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level1(src)
+			revenant_level = 1
+			revenant_damage = 25
+	if(revenant_level == 1)
```

Comment:
I would make these "else if" as if you go into any of them there's no need to check the others afterwards